---
title: "Alpha Diversity Demystified: Part 1"
author: Emily Van Syoc
date: '2022-03-10'
slug: alpha-div-part1
categories: [tutorials]
tags: [alpha-div]
subtitle: 'Part 1: What is alpha diversity?'
summary: 'Learn the basics of richness and evenness with a simplified microbiome'
authors: []
lastmod: '2022-03-10T21:24:34-05:00'
featured: no
share: true
commentable: true
image:
  caption: ''
  focal_point: ''
  preview_only: no
bibliography: biblio.bib
projects: []
output:
  blogdown::html_page:
    toc: true
    #toc_float: true
    fig_width: 6
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

# What is alpha diversity

If you have read a microbiome paper, chances are you've come across the terms "alpha diversity" and "beta diversity" - sometimes also called "within-sample diversity" and "between-sample diversity", respectively. These terms were coined by Robert Whittaker in 1972 and have become ubiquitous in microbiome research [@whittaker1972]. 

Alpha diversity means two things: *richness* and *evenness*. However, we don't usually use those terms directly; instead, alpha diversity is calculated in different indices that consider richness, evenness, or both. You've probably seen these indices before but may not know how they differ.

So, **how do we choose the appropriate alpha diversity index for our analysis?** 


## How do we understand alpha diversity indices?

We'll create a simplified microbiome for two fake people, and dig all the way into `vegan` source code to calculate these indices by hand (it's not as scary as it sounds!). Then, we'll discuss the differences between commonly used alpha diversity indices, how to apply appropriate statistical analyses, and if they violate the concept of compositional analysis (spoiler alert...). 

## What you need to follow this guide

Ideally, you will have at least basic experience in R. My code is written with tidy methods and we will start with a phyloseq object. If you're following along in your own R session, install these packages first using the code chunk below.


```{r, eval = FALSE, code_folding = TRUE}

# if needed, install the requisite packages

# CRAN installs
install.packages(c("vegan", "ggpubr", "BiocManager", "tidyverse"))

# Bioconductor installs
BiocManager::install("phyloseq")

```

# Creating a simplistic data set

Let's create an overly simplistic microbiome for two fake people, Jenny and Sarah. There are 3 bacteria in their microbiome; Taxa A, Taxa B, and Taxa C.


```{r}

# first, load the requisite packages
library(phyloseq)
library(vegan)
library(tidyverse)
library(ggpubr)

# create our dataset
samples <- data.frame(
  row.names = c("Jenny", "Sarah"),
  taxaA = c(1, 2),
  taxaB = c(3, 11),
  taxaC = c(5, 0)
)

# convert the dataframe to a phyloseq object
otu <- phyloseq::otu_table(samples, taxa_are_rows = FALSE)

```

# Estimate alpha diversity in phyloseq

Phyloseq has a function called `estimate_richness()` that calculates commonly used alpha diversity indices. By default, it returns Observed Taxa, Chao1, se.chao1, ACE, se.ACE, Shannon, Simpson, Inverse Simpson, and Fisher. 

If we look at the help page for `estimate_richness()`, it shows us that the internal functions are from `vegan`: `estimateR`, `diversity`, and `fisherfit`. The `estimateR` help page shows the equations used for Chao1 and ACE, and `diversity` shows the equations for Shannon, Simpson, and Inverse Simpson. `fisherfit` uses a model fit function from `fisher.alpha` which is a little complex for this guide. We'll focus on the indices that are easily calculated from the OTU table input.

```{r, eval = FALSE}

# open help page for estimate_richness
?phyloseq::estimate_richness

# open help page for estimateR
?vegan::estimateR

# open help page for diversity
?vegan::diversity

```

``` {r, layout = "l-body-outset"}

# estimate alpha diversity
alpha <- phyloseq::estimate_richness(otu)

# visualize results
knitr::kable(alpha)

```

# Richness indices: Observed taxa, Chao1, and ACE

"Richness" can be thought of as simply "the number of species". These indices emerged for field biologists to quantify the number of species in a sample [@ohara2005]. You can think of this as counting the number of tigers in a jungle.

Observed Taxa is straightforward - it is the absolute richness value, which is simply the number of taxa detected in a sample. Chao1 and ACE are **non-parametric estimators** that derive the "true" richness in a sample from an equation. These functions are estimators that rely on one simple assumption: the true number of tigers present in the jungle is always higher than the number of tigers you have actually managed to capture. Anne Chao called this "capture heterogeneity" [@chao1987]. Chao1 and ACE both estimate the number of "rare" species in order to extrapolate the true richness.

*This is why Chao1 and ACE have standard error estimates but Observed Taxa does not: Observed Taxa is a count, while Chao1 and ACE are estimators.*

## Observed Taxa

"Observed Taxa" is also called "number of species" or simply "richness". It is exactly what it sounds like - the number of species detected in a sample. 


```{r}
## calculate Observed Taxa for Jenny

# first, subset our dataframe to Jenny's microbiome
jenny <- samples[1, ]

# then, calculate the total number of taxa present in her microbiome
(tot.taxa = length(jenny[jenny > 0])) # print

# that's it! That is the Observed Taxa

```

If we plot Jenny's Observed Taxa against Sarah's, we conclude that Jenny has a higher number of species in her microbiome. In fact, we could even say that Jenny has *1.5 times* the number of bacteria in her microbiome!

```{r}

## plot the Observed Taxa for both samples

# first, make a column for Sample
plot.dat <- alpha %>% 
  rownames_to_column(var = "Sample")

# plot
ggscatter(data = plot.dat, x = "Sample", y = "Observed", 
          size = 3,
          color = "Sample",
          subtitle = "Jenny has 1.5 times the number of taxa as Sarah")


```


### What is the problem here?

There's some issues to contend with, mostly dealing with the idea of sequencing depth and compositionality. Microbiome data comes from a sequencer, which has as limited number of slots. You can think of sequencing data as a bunch of little bugs vying for a spot on the sequencing bus. Only so many bugs will get a seat! Therefore, **we can't conclude if Jenny's microbiome truly has more bacterial species than Sarah's, or if the species in Sarah's microbiome are rarer and were not sequenced**. Additionally, samples with deeper sequencing are more likely to have more taxa detected.

Also, keep in mind that richness doesn't tell us anything about **abundance** or **functionality**. 

Finally, the effect size here is important. We may see a 1.5 fold change between Jenny and Sarah, but the reality is that they differ by 1 species. Is that a biologically meaningful effect size?

## Chao1

Anne Chao described an equation for richness in 1987 for field biologists, and the equation has evolved in the decades since [@chao1987]. The general idea of this equation is that in a closed population, capture probability is heterogeneous; that is, not all capture times and capture locations result in the same number of captures. Most importantly, there will be a number of species that remain uncaptured. Chao1 seeks to estimate the "true" richness of a sample given this assumption.

How does this translate to microbial ecology? Chao1 uses the "frequency of species" to extrapolate richness. This is different than Observed Taxa because in Chao1, *we are estimating the "true" richness given the parameters*. This is why Chao1 has a standard error estimate (which `phyloseq` and `vegan` prints as `se.Chao1`).

The equation for Chao1 in the `estimateR` help page is actually slightly different than the equation in the `vegan` source code. In the source code, the Chao1 equation is:

$$
Chao1 = S_0 + SSC * \frac{a1 * (a1 - 1)}{\frac{a2 + 1}{2}} 
$$
where $S_0$ is the total observed species, $SSC$ is a "small sample correction", $a1$ is the number of singletons (taxa observed only once), and $a2$ is the number of doubletons (taxa observed only twice).


In the code chunk below, the Chao1 equation is taken directly from the [vegan source code](https://github.com/vegandevs/vegan/blob/master/R/estimateR.default.R). We'll calculate Chao1 by hand for Jenny's microbiome and compare this to the `phyloseq` output.

```{r}

## calculate CHAO1 for Jenny

# first, subset our dataframe to Jenny's microbiome
jenny <- samples[1,]

## ---- BUILD CHAO1 EQUATION 

# get total number of species
S_0 = length(jenny[jenny > 0])

# get number of singletons
a1 = length(which(colSums(jenny) == 1))

# get number of doubletons
a2 = length(which(colSums(jenny) == 2))

## calculate chao1 using the equation from vegan source code
chao1 = S_0 + 1 * a1 * (a1 - 1) / (a2 + 1) / 2

# does this match what phyloseq calculated?
identical(chao1, alpha$Observed[1])

```

Since our fake microbiome data is very small, there are no differences between Chao1 and Observed Taxa. We still conclude that Jenny has a higher richness than Sarah.

## ACE

The ACE index, or "Abundance Coverage Estimator", relies on the same assumption as Chao1; that in a community of a fixed size, we can only sample so many species and must extrapolate to quantify the true richness of the community [@ohara2005]. ACE is a variant of Chao1. 

Here's one key difference: **ACE sets an arbitrary threshold to determine which species are considered "rare"**. In `vegan`, this threshold is 10. This means that any taxa with abundance less than or equal to 10 is considered "rare", and any species with abundance over 10 is considered "abundant".

Here is the ACE equation from the [vegan source code](https://github.com/vegandevs/vegan/blob/master/R/estimateR.default.R):

$$
ACE = S.abund + \frac{S.rare}{C.ace} + max(\gamma, 0) * \frac{a1}{C.ace}
$$
Where $S.abund$ is the number of "abundant" taxa, $S.rare$ is the number of "rare" taxa, and $N.rare$ is the number of individuals in rare taxa. $C.ace$ is defined as $1-\frac{a1}{N.rare}$, and $\gamma$ is a function of the sum of taxa occurrences up to 10, S.rare, and C.ace.

```{r}

## calculate ACE for Sarah

# first, get Sarah's microbiome
sarah <- samples[2,]

## ---- Build ACE equation 

# get number of "rare" taxa (under or equal to 10)
S_rare = length(which(sarah[sarah > 0] <= 10)) ## NOTE: does NOT include zero abundance (taxa which are present in Jenny's microbiome but not Sarah's)

# get number of "abundant" taxa (over 10)
S_abund = length(which(sarah[sarah > 0] > 10))

# get number of singletons
a1 = length(which(colSums(sarah) == 1))

# get number of individuals in rare species
N.rare = sum(sarah[sarah < 11])

# get C.ace
C.ace = 1 - a1 / N.rare

### "Gamma" model from vegan source code
i <- 1:10
COUNT <- function(i, counts) {
  length(counts[counts == i])
}
a <- sapply(i, COUNT, sarah) # this counts the number of occurrences of taxa up to 10
i <- seq_along(a)
thing <- i * (i - 1) * a
Gam <- sum(thing) * S_rare/(C.ace * N.rare * (N.rare - 1)) -  1


## calculate ACE
ACE <- S_abund + S_rare/C.ace + max(Gam, 0) * a1/C.ace

# does this match what phyloseq calculated?
identical(ACE, alpha$ACE[2])

```

Similarly to Chao1, we conclude that Jenny has higher richness than Sarah.

### Is this problematic?

Does the arbitrary threshold of 10 scale to different ecological systems? Keep in mind that these indices were created several decades before next-generation sequencing allowed us to capture thousands of bacteria in a single sample. Is a taxa's abundance considered "rare" if there are fewer than 10 counts in a sample? If our sequencing depth captured 300,000 reads in Sarah's microbiome, then maybe we consider an abundance of 3,000 to be "rare" (0.01%). When using the ACE equation, keep this threshold in mind. 

## Comparing richness estimates

Let's compare our three estimates of richness for our fake people Jenny and Sarah.

```{r}

## plot all 3 estimates of richness

# first, make vertical
plot.datv <- plot.dat %>% 
  pivot_longer(cols = c("Observed", "Chao1", "ACE"), 
               names_to = "index", 
               values_to = "richness") %>% 
  # round our values to plot them clearly
  mutate(richness = round(richness, 2))

# plot
ggbarplot(data = plot.datv, x = "Sample", y = "richness", fill = "index",
          # group indices together 
          group = "index", position = position_dodge(0.8), label = TRUE,
          # move legend to left of the plot
          legend = "left",
          # add subtitle
          subtitle = "All richness estimates") 
```

Importantly, we would come to the same conclusion regardless of the index we chose for our analysis: Jenny's microbiome has higher richness than Sarah's. We would observe a slightly larger effect size with ACE, but it would not change our overall results.


# Evenness indices: Shannon's, Simpson's, and Inverse Simpson's

Observed Taxa, Chao1, and ACE are all estimates of richness, or the number of taxa. Shannon's, Simpson's, and Inverse Simpson's quantify richness in addition to "evenness", or the distribution of the species abundance. Evenness can be thought of the balance of each taxa present in a sample: If Jenny has 3 bacteria in her microbiome and each has an abundance of 50, then she has perfect evennness. If she has 3 bacteria in her microbiome and one has an abundance of 50, one has an abundance of 1500, and one has an abundance of 2, then she has lower evenness. In the field of population ecology, a population is considered less diverse if it is dominated by one or two species, and considered more diverse if the abundance of each taxa is distributed evenly.

## Shannon's index

The Shannon index (also called Shannon-Wiener Index or Shannon's H) is perhaps the most commonly used index of alpha diversity. It was derived in the 1940's by Claude Shannon and has become ubiquitous in most life science disciplines [@spellerberg2003]. In technical terms, Shannon's diversity is a calculation for "entropy"; it is the uncertainty of predicting which species the next sample is.

Let's take a quick side step to illustrate this: imagine two ball pits. Ball pit #1 contains 9 blue balls and 1 red ball. What is the probability that the next ball we randomly select from the pit will be red? 

Ball pit #2 contains 4 blue balls and 5 red balls. Now how certain can we be that the next ball we randomly select will be red? Ball pit #2 has the same **richness** as Ball pit #1, but it is more **even** and will therefore have a higher Shannon's H. 

Calculating Shannon's H by hand can help us contextualize this in a microbiome analysis. The equation is:

$$ 
H = -\sum_{i = 1}^Np_i * log(b)p_i
$$

Where $p_i$ is the proportional abundance of taxa $i$ and $b$ is the log base. The default log base in `phyloseq` and `vegan` is $exp(1)$. 


```{r}

## calculate Shannon's H for Sarah

# first, get a sum of Sarah's taxa for proportional abundance
sum.taxa <- sum(sarah)

# the default log parameter in phyloseq is exp(1)
b = exp(1)

# calculate Shannon's H
H = -sum(
  # taxa A
  ((sarah[,1] / sum.taxa) * log((sarah[,1] / sum.taxa), base = b)),
  # taxa B
  ((sarah[,2] / sum.taxa) * log((sarah[,2] / sum.taxa), base = b))
  # taxa C has abundance ZERO
)

# compare to phyloseq's calculation
identical(H, alpha$Shannon[2])

```


## Simpson's index

Simpson's index is very closely related to Shannon's index and was also derived in the 1940's by E.H. Simpson [@simpson1949]. We're still working with proportional abundances of each taxa. Simpson's index is simply the sum of the squared proportional abundances.

First, we calculate "Simpson's concentration", or *D*:

$$
D = \sum_{i = 1}^N p_i^2
$$

Then, Simpson's index is derived as: $$Simpsons = 1-D$$.


What is $D$? This is the probability that two random draws from the same sample belong to the same species [@lande1996]. Let's go back to our ball pit from Shannon's diversity. If Ball pit #1 has 9 blue balls and 1 red ball, the probability that two random draws result in two blue balls are high. Ball pit #2 contains 4 blue balls and 5 red balls. What is the probability that two random draws will result in two blue balls? We are much less confident in these odds. 

Simpson's index is the probability that two random draws belong to **different** species. 

```{r}

## calculate Simpson's index for Sarah

# first, get a sum of Sarah's taxa for proportional abundance
sum.taxa <- sum(sarah)

# calculate Simpson's index D
D = sum(
  # squared proportional abundance of taxa A
  (sarah[,1] / sum.taxa) ^ 2,
  # squared proportional abundance of taxa B
  (sarah[,2] / sum.taxa) ^2,
  # squared proportional abundance of taxa C
  (sarah[,3] / sum.taxa) ^ 2
)

# return Simpson's index
Simpson = 1-D


# compare to phyloseq's calculations
identical(Simpson, alpha$Simpson[2])

```

## Inverse Simpson

Inverse Simpson is simple to calculate. It is quite literally the inverse of Simpson's diversity. 

First, we start with *D* from above:
$$
D = \sum_{i = 1}^N p_i^2
$$
Then, we calculate the inverse: $$Inverse Simpson = 1/D$$. 

```{r}

# calculate Inverse Simpson for Sarah
InvSimp = 1/D

# compare to phyloseq's calculations
identical(InvSimp, alpha$InvSimpson[2])

```

### What's the difference between Inverse Simpson and Simpson?

Inverse Simpson is calculated in such a way that in a community with total number of species $S$, the maximum value of Inverse Simpson's is equal to $S$ in a perfectly even community (all species have the same proportional abundance) [@lande1996]. 


## Comparing evenness estimates

```{r}

## compare all three evenness estimates

# first, make vertical
plot.datv <- plot.dat %>% 
  pivot_longer(cols = c("Shannon", "Simpson", "InvSimpson"), 
               names_to = "index", 
               values_to = "richness") %>% 
  # round our values to plot them clearly
  mutate(richness = round(richness, 2))

# plot
ggbarplot(data = plot.datv, x = "Sample", y = "richness", fill = "index",
          # group by index
          group = "index", position = position_dodge(0.8), 
          # add the estimate to the top of the bars
          label = TRUE,
          # move legend to left
          legend = "left",
          # add subtitle
          subtitle = "All evenness indices") 
```

Similarly to the richness estimates, our general conclusions are the same regardless of which index we choose. We conclude that Jenny's microbiome has higher "alpha diversity" than Sarah's microbiome. 

## Which one do I choose?

I hope that I've convinced you of a distinction between richness indices and evenness indices. Depending on your research question, it may be appropriate to evaluate one measure of richness and one measure of evenness. Or, you may only be interested in the evenness of your system (after all, most interventions to change the microbiome aren't expected to remove or add species and therefore we won't expect to see a difference in richness).

However, the `vegan` help page for `diversity` says it best: 

> ...However, these indices are all very closely related (Hill 1973), and there is no reason to despise one more than others (but if you are a graduate student, don't drag me in, but obey your Professor's orders). 

## Next: Put this into practice!

Join us in [Part 2](post/alpha-diversity-part2) to perform a statistical analysis on a "real" microbiome! 